packages/astro/src/core/build/static-build.ts

```diff
@@ -424,7 +514,7 @@ async function copyFiles(fromFolder: URL, toFolder: URL, includeDotfiles = false
 			const lastFolder = new URL('./', to);
 			return fs.promises
 				.mkdir(lastFolder, { recursive: true })
-				.then(() => fs.promises.copyFile(from, to));
+				.then(() => fs.promises.copyFile(from, to, fs.constants.COPYFILE_FICLONE));
```

Comment:
Small perf optimization: we can use [copy on write](https://en.wikipedia.org/wiki/Copy-on-write) if available. Node automatically uses a fallback on unsupported systems.

Reply:
macOS people stay winning (and bearded people on Linux, I guess)